elastic/apm-server
# Update apm-data: set a default `span.representative_count`

## Motivation/summary

Update to apm-data changes:
 - https://github.com/elastic/apm-data/pull/30 (`span.representative_count` now defaults to 1)
 - https://github.com/elastic/apm-data/pull/33 (removes `modelprocessor.DroppedSpansStatsDiscarder`, now filtered in encoding)
 - https://github.com/elastic/apm-data/pull/35 (overhauled JSON encoding approach)

The metrics data structure has changed, so a new common ingest pipeline is added to handle these. We now send metrics as 

```json
"metricset.samples": [
  {"name": "metric_name1", "type": "histogram", "unit": "s", "values": [1, 2, 3], "counts": [4, 5, 6]},
  {"name": "metric_name2", "type": "summary", "value_count": 123, "sum": 456},
  {"name": "metric_name3", "type": "counter", "value": 123}
  {"name": "metric_name4", "value": 456}
]
```

The ingest pipeline takes care of mapping these to top-level metric fields as before.

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

The only functional change should be the `span.representative_count` one. This can be tested by sending spans, with destination fields set, to the server without `sample_rate` set, and checking that (a) the span documents indexed have `representative_count: 1`, and (b) service_destination metrics are produced.

## Related issues

Closes https://github.com/elastic/apm-server/issues/3565
